models/org_team.go

```diff
@@ -324,6 +348,14 @@ func NewTeam(t *Team) (err error) {
 		}
 	}
 
+	// Add all repositories to the team if it has access to all of them.
+	if t.IsAllRepositories {
+		err = t.addAllRepositories(sess)
```

Comment:
We only need a flag on team but unnecessary really to add all repositories to the team I think.

Reply:
That would complicate things, currently imho it would be ok if it would work just like owners team

Reply:
@lafriks yes, that's what my opinion. We don't need save all the repositories on this team, because maybe somehow a new repository will be created on this organization.

Reply:
@lunny that could be done in other PR as that could be quite large change as would require changes in all places where right check is loaded

Reply:
@lunny I don't understand you're remark:

> because maybe somehow a new repository will be created on this organization

Any new repository created in the organization will be added to the team as it is for the Owner team:
https://github.com/go-gitea/gitea/blob/f6983856de8a534ff23affd7c09006d7195974b0/models/repo.go#L1358-L1369

maybe I missed something.

Reply:
@lunny @lafriks 
I looked at lunny's proposal to see how to do it.

IMHO even if we don't add all repositories on the teams, I think the team's API should be consistent for a team with tag `IncludesAllRepositories` setted:
- `Team.GetRepositories`: should load all organization repositories in `Team.Repos`
- `Team.HasRepository`: should return true if it is a repository of the organization
- `Team.NumRepos`: this field should be equals to the number of repositories in the organization

The two functions `Team.AddRepository` and `Team.RemoveRepository` will be no-op func. They will do nothing.

There will be some changes in `NewTeam`, `UpdateTeam`, `DeleteTeam`, `createRepository` and `deleteRepository` to update the field `Team.NumRepos`, calculate the team's members authorizations and manage watches (repo and issues).
I do not detail the changes here. They are not so large. I could if you're interested in.

Does this meet your expectations?
Do you have any comments, suggestions, improvements?

Thanks for your feedback.

Reply:
I mean if a team has `IncludesAllRepositories = true`, it should have the same behavior on operating repositories with owners team. So I don't think there are so many changes. We only need to change check `isOwnerTeam` to `IncludesAllRepositories == true`, that should be work.

Reply:
I'm not sure I fully understand.
We do not want to add all repositories to the teams, right?
But currently, Owner team has all repositories.

Reply:
@ngourdon is right owner teams also currently has all repositories added